osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -438,7 +441,7 @@ private void updateSelectedBeatmap(BeatmapInfo beatmap)
 
         private void updateSelectedRuleset(RulesetInfo ruleset)
         {
-            if (ruleset?.Equals(rulesetNoDebounce) == true)
+            if (ruleset == null || ruleset.Equals(rulesetNoDebounce))
```

Comment:
Not sure why this doesn't mirror the `updateSelectedBeatmap` check. Seems like `rulesetNoDebounce` being `null` is also valid (`performUpdateSelected` is at least one place that seems to permit it - it is assigned to a `ruleset` local and then logged).

Also not sure why this doesn't fail any tests; it's probably because `rulesetNoDebounce` seems to have no functional usages besides that one log statement. Still would probably unify treatment with beatmap to reduce confusion/avoid potential future bugs.

Reply:
I've made this change 👍🏻